operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/SSABasedGenericKubernetesResourceMatcher.java

```diff
@@ -104,12 +104,23 @@ public boolean matches(R actual, R desired, Context<?> context) {
     removeIrrelevantValues(desiredMap);
 
     if (LoggingUtils.isNotSensitiveResource(desired)) {
-      log.debug("Pruned actual: \n {} \n desired: \n {} ", prunedActual, desiredMap);
+      logDiff(prunedActual, desiredMap, objectMapper);
     }
 
     return prunedActual.equals(desiredMap);
   }
 
+  private void logDiff(Map<String, Object> prunedActualMap, Map<String, Object> desiredMap,
+      KubernetesSerialization serialization) {
+    if (log.isDebugEnabled()) {
+      var actualYaml = serialization.asYaml(prunedActualMap);
+      var desiredYaml = serialization.asYaml(desiredMap);
+      log.debug("Pruned actual yaml: \n {} \n desired yaml: \n {} ", actualYaml, desiredYaml);
+    } else {
+      log.debug("Pruned actual map: \n {} \n desired map: \n {} ", prunedActualMap, desiredMap);
```

Comment:
Not sure if I follow this. "If debug is not enabled we log something on debug" ?

Reply:
Ah yes, missed that, though there should actually be no output since `debug` is checked internally (iirc) but it does make sense to remove the else branch altogether, indeed.

Reply:
What I mean is that the else block never logs any message, since only called if no debug is enabled but logs on debug level.

Reply:
@csviri 

Ah understand

https://github.com/operator-framework/java-operator-sdk/pull/2542/commits/96920abbe7322c0fa20f36c22065ebec8c067b87

I thought it would be better to just remove this part, so I modified it as shown above. What do you think?